python/cudf/utils/cudautils.py

```diff
@@ -610,6 +610,39 @@ def gpu_diff(in_col, out_col, N):
             out_col[i] = -1
 
 
+@cuda.jit
+def gpu_round(in_col, out_col, decimal):
+    i = cuda.grid(1)
+    round_val = 10 ** (-1.0 * decimal)
+
+    if i < in_col.size:
+        if not in_col[i]:
+            out_col[i] = np.nan
+            return
+
+        newval = in_col[i] // round_val * round_val
+        remainder = fmod(in_col[i], round_val)
+
+        if remainder != 0 and remainder > (.5 * round_val) and in_col[i] > 0:
+            newval = newval + round_val
+            out_col[i] = newval
+
+        elif remainder != 0 and abs(remainder) < (.5 * round_val) and \
+                in_col[i] < 0:
+            newval = newval + round_val
+            out_col[i] = newval
+
+        else:
+            out_col[i] = newval
+
+
+def apply_round(data, decimal):
+    output_dary = rmm.device_array_like(data)
+    if data.size > 0:
```

Comment:
Since you're configuring the kernel against the `output_dary`, should be consistent.
```suggestion
    if output_dary.size > 0:
```